include/examples/simple_io_behavior.h

```diff
@@ -0,0 +1,146 @@
+   */
+  using kodlab::IOBehavior<SimpleRobot, PDGains, MotorLog>::robot_;
+
+  /**
+   * @brief Using the default `IOBehavior` constructor
+   */
+  using kodlab::IOBehavior<SimpleRobot, PDGains, MotorLog>::IOBehavior;
+
+  /**
+   * @brief Initialize class members and return true upon completion
+   * @return true if initialization successful, false otherwise
+   */
+  bool Init() override {
+    // Initializes desired torques to zero
+    desired_torques_ = std::vector<float>(robot_->joints.size(), 0);
+
+    // Initialize joint positions and velocities
+    joint_positions_ = robot_->GetJointPositions();
+    joint_velocities_ = robot_->GetJointVelocities();
+
+    // Return true to indicate successful initialization
+    return true;
+  }
+
+  /**
+   * @brief Begin the behavior
+   * @details Record the current position as a set point and start the internal
+   * behavior timer from zero.
+   * @param prev_behavior the previously active behavior
+   */
+  void Begin(const kodlab::Behavior<SimpleRobot> &prev_behavior) override {
+    // Begin realtime timer
+    time_now_ = 0.0;
+    timer_.tic();
+
+    // Set desired position as current position
+    desired_position_ = robot_->GetJointPositions()[0];
+  }
+
+  /**
+   * @brief Compute the PD-controller update and set robot torques
+   * @details This method updates the behavior timer, retrieves the latest joint
+   * state, and uses this information to compute desired control input via a
+   * proportional-derivative scheme using the member gains.
+   */
+  void Update() override {
+    // Update time
+    time_now_ = timer_.tac();
+
+    // Retrieve current joint positions
+    joint_positions_ = robot_->GetJointPositions();
+    joint_velocities_ = robot_->GetJointVelocities();
+
+    // Compute command torque for joint 0 via PD, set all others to zero
+    desired_torques_ = std::vector<float>(robot_->joints.size(), 0.0);
+    float pos_error = desired_position_ - joint_positions_[0];
+    float vel_error = DESIRED_VELOCITY - joint_velocities_[0];
+    desired_torques_[0] = gains.kp * pos_error + gains.kd * vel_error;
+
+    // Set joint torques
+    robot_->SetTorques(desired_torques_);
+  }
+
+  /**
+   * @brief Process new input data on receipt
+   * @details Overwrite the member gains with the control torques received via
+   * the LCM input.
+   * @param input_msg newly received input message
+   */
+  void ProcessInput(const PDGains &input_msg) override {
+    gains.kp = input_msg.kp;
+    gains.kd = input_msg.kd;
+    std::fprintf(stdout,
+                 "[INFO][SimpleIOBehavior] Setting kp = %.3f, kd = %.3f\n",
+                 gains.kp,
+                 gains.kd);
+  }
+
+  /**
+   * @brief Prepare output log for publishing
+   * @details Populate the fields of a MotorLog message with the controlled
+   * joint's state.
+   * @param output_msg message to be published
+   */
+  void PrepareOutput(MotorLog &output_msg) override {
+    output_msg.timestamp = time_now_;
```

Comment:
Does this log no longer have the timing information?

Reply:
The `MotorLog` LCM message in particular does not have margin or message duration in it.  Honestly, I don't think the timing included here is particularly useful, but I wanted to demonstrate how a behavior clock could be made and used to log behavior-specific information.  In my opinion, we should treat the control loop clock as ground truth, and pass the timer around to the various classes that will be logging so that they can include a ground truth timestamp in their log messages.